### PR TITLE
auto link nodes when created with menu

### PR DESCRIPTION
**Pull request by asylves1**
# Description

When node is created using the node menu auto connect the input and output

Resolves #(issue)

### CODE REVIEW

File: packages/client/hmi-client/src/components/workflow/tera-workflow.vue [hunk 1]
```diff
@@ -465,7 +482,15 @@ function onMenuSelection(operatorType: string, menuNode: WorkflowNode<any>) {
...
 	if (name && operation && node && drilldown) {
-		addOperatorToWorkflow({ name, operation, node, drilldown })();
+		const newNode: WorkflowNode<any> = addOperatorToWorkflow({ name, operation, node, drilldown })();
+		const inputPort = newNode.inputs.find((input) => input.type.includes(port.type));
+
+		if (inputPort) {
+			// wait for the DOM to load the new node before adding the edge
+			await nextTick();
+			createNewEdge(menuNode, port, direction);
+			createNewEdge(newNode, inputPort, 0);
```
Comment (mwdchang) [inline packages/client/hmi-client/src/components/workflow/tera-workflow.vue:492]: `createNewEdge` is meant for interactive use - when modellers are dragging stuff. 

If you want to "auto connect" use `wf.value.addEdge(...)` instead, it is more straightforward and you don't need to deal with "newEdge" transient state.

File: packages/client/hmi-client/src/components/workflow/tera-workflow.vue [hunk 2]
```diff
@@ -465,7 +474,18 @@ function onMenuSelection(operatorType: string, menuNode: WorkflowNode<any>) {
 	newNodePosition.y = menuNode.y;
 
 	if (name && operation && node && drilldown) {
-		addOperatorToWorkflow({ name, operation, node, drilldown })();
+		const newNode: WorkflowNode<any> = addOperatorToWorkflow({ name, operation, node, drilldown })();
+		const inputPort = newNode.inputs.find((input) => input.type.includes(port.type));
+
+		if (inputPort) {
+			// wait for the DOM to load the new node before adding the edge
+			await nextTick();
```
Comment (shawnyama) [inline packages/client/hmi-client/src/components/workflow/tera-workflow.vue:482]: Now that you're adding the edge differently, I wonder if that `nextTick()` is required anymore

File: packages/client/hmi-client/src/components/operator/tera-operator.vue [hunk 3]
```diff
@@ -114,8 +114,8 @@ function mouseleavePort() {
 	emit('port-mouseleave');
 }
 
-function onSelection(operatorType: string) {
-	emit('menu-selection', operatorType);
+function onSelection(operatorType: string, input: WorkflowPort) {
```
Comment (shawnyama) [inline packages/client/hmi-client/src/components/operator/tera-operator.vue:117]: `input` should be named `output` or `outputPort`

**Review by shawnyama — APPROVED** [review]

File: packages/client/hmi-client/src/components/operator/tera-operator.vue [hunk 4]
```diff
@@ -36,7 +36,7 @@
 		<tera-operator-outputs
 			:outputs="node.outputs"
 			:menu-options="menuOptions"
-			@menu-selection="onSelection"
+			@menu-selection="(operatorType: string, input: WorkflowPort) => onSelection(operatorType, input)"
```
Comment (shawnyama) [inline packages/client/hmi-client/src/components/operator/tera-operator.vue:39]: should be `outputPort` here too

File: packages/client/hmi-client/src/components/workflow/tera-workflow.vue [hunk 5]
```diff
@@ -465,7 +474,18 @@ function onMenuSelection(operatorType: string, menuNode: WorkflowNode<any>) {
 	newNodePosition.y = menuNode.y;
 
 	if (name && operation && node && drilldown) {
-		addOperatorToWorkflow({ name, operation, node, drilldown })();
+		const newNode: WorkflowNode<any> = addOperatorToWorkflow({ name, operation, node, drilldown })();
+		const inputPort = newNode.inputs.find((input) => input.type.includes(port.type));
```
Comment (mwdchang) [inline packages/client/hmi-client/src/components/workflow/tera-workflow.vue:478]: This check will not work complex types, for example: `modelConfigId|datasetId` linking to `modelConfigId|modelId`

Comment (mwdchang) [reply packages/client/hmi-client/src/components/workflow/tera-workflow.vue:478]: Also note the menuItem generation is incorrect .... I will see about creating a PR to fix that front.

Comment (mwdchang) [reply packages/client/hmi-client/src/components/workflow/tera-workflow.vue:478]: See https://github.com/DARPA-ASKEM/terarium/pull/4502

File: packages/client/hmi-client/src/components/workflow/tera-workflow.vue (hunk 2 above)
Comment (asylves1) [reply packages/client/hmi-client/src/components/workflow/tera-workflow.vue:482]: looks like it's still needed :(

**Pull request merged**